src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -0,0 +1,144 @@
+"""
+    Copyright 2022 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import abc
+import logging
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import Dict, List
+
+from inmanta.data import EnvironmentMetricsCounter, EnvironmentMetricsNonCounter
+from inmanta.protocol.exceptions import ServerError
+from inmanta.server import SLICE_DATABASE, SLICE_ENVIRONMENT_METRICS, SLICE_TRANSPORT, protocol
+
+LOGGER = logging.getLogger(__name__)
+
+COLLECTION_INTERVAL_IN_SEC = 60
+
+
+class MetricType(str, Enum):
+    """
+    There are 3 types of metrics: metrics that represent a counter, metrics that doesn't represent a counter and
+    the compile_rate metric which is a special case. An examples of the count type metric is the agent_count metric which counts
+    the number of agents at a timestamp. An example of a non_count metric is the compile_time metric. This metric counts the
+    numer of compiles between the current timestamp and the last one and the total compile time.
+    The compile_rate metric is a special case as it is being calculated using only the count field of the compile_time metric.
+    """
+
+    COUNT = "count"
```

Comment:
I think gauge, timer and rate (which is derived from timer)  are the proper names, if we follow the terminology of https://github.com/omergertel/pyformance (which we already use)

Reply:
isn't the rate called meter in this terminology?

Reply:
well, if it is the only thing you measure: yes